src/mixins/name-request-mixin.ts

```diff
@@ -64,8 +64,7 @@ export default class NameRequestMixin extends Mixins(DateMixin) {
     return {
       // workaround for old or new property name
       nrNumber: nr.nrNum || nr.nrNumber,
-      // FUTURE: Update entityType to use nr.requestTypeCd when namex supports our entity types
-      entityType: CorpTypeCd.BENEFIT_COMPANY,
+      entityType: nr.legalType,
```

Comment:
Don't we have a table or something to convert from "type in NR" to "type in Entity" ?

Reply:
Or are the types the same between namex and Lear?

Reply:
Also, what's the difference between `legalType` and `requestTypeCd` ?

Reply:
ah you are right, there could be a conversion mismatch for BC. I forgot, good catch!

Reply:
Or was it CR?

Reply:
Lol, CR (in NR) is BC in Lear, so we're both right.

Reply:
the legalType is the entityType and the requestTypeCd as namex calls it, is the code for describing what the user wants to use the NR for afaik.
ie new business, restoration change of name etc.

Reply:
https://github.com/bcgov/bcrs-shared-components/blob/06785da8bbeb1594100243a30d5639d60e0952df/src/modules/corp-type-module/corp-type-module.ts#L19

Reply:
I'm wrong, i'm not actually sure..

Reply:
> the legalType is the entityType and the requestTypeCd as namex calls it, is the code for describing what the user wants to use the NR for afaik.
> ie new business, restoration change of name etc.

I looked at some sample NRs and both were the same value 🤷‍♂️

Reply:
oh no, i was right. The `requestTypeCd` is the code that existed to describe the combination of entity type, request type AND location. 
So these codes mapped to a specific type of NRs. It just so happens that the legalType code often matches the request Type code for basic NEW business requests of specific entity types, in this case a COOP.

Clear as mud eh hahaha
Basically a legacy code.

Reply:
Should this be updated too?
https://github.com/bcgov/business-filings-ui/blob/da20983945e2da3e5d1bcadbcb10b83991cbf921/src/App.vue#L603

Reply:
I'm not sure, i don't believe `requestTypeCd` and `entityType` are always the same thing. I'm pretty sure if we look at requestTypeCd mappings in Namex they will describe a combination of values, where entity type is just entity type. 
I'm not too sure about uncommenting this at the moment.

Reply:
After looking at this further, we are just parsing this NR for display purposes and the conversion IS happening in computed value `entityTypeDescription` :) 
Good for bringing up though, had to check!

Reply:
OK, so requestTypeCd is different from entityType.

Should that check in App.vue be uncommented, using `nr.legalType` ?

The original idea was to verify that the NR type matches the entity type. Is this guaranteed some other way? If not then this check should be enabled.

Reply:
^^ If the the check in App.vue is not needed then please delete it (and add a comment indicate why it's not needed). Eg,
> don't need to verify NR type because ...

Reply:
Hmm i really don't know. Not sure i'm comfortable with either at this point. Would need to look at it for bit longer than i have time for at this moment.

Reply:
OK, go with what you have now (plus CR/BC conversion), test it well, and then make a note to look at it later.

Reply:
That conversion is happening prior to this point, can confirm. When we pass in the NR to the mixin, it's already reflecting the correct type. 
ie a Benefit Company NR(BC) is of the correct Legal Type for Lear (BEN) prior to the point of converting to the display name.